### PR TITLE
Do not log OperationCanceledException on user cancel (fixes #65)

### LINKED ISSUE #65: "Unhandled event loop exception" on "Cancel user task"
- Tried to edit a file during the long running build.
- Blocking progress dialog shows me, that I have to wait.
- I don't wait so click on "Cancel user operation"
- "Unhandled event loop exception" error logged to the error log

This is coming from https://bugs.eclipse.org/bugs/show_bug.cgi?id=577511 / https://github.com/eclipse-platform/eclipse.platform.text/commit/0f409d6f3d6b590242268c9d97c3a0107737dfa7 / https://github.com/eclipse-platform/eclipse.platform.text/pull/4.

If we re-throw this exception in order to fix something, we should not log it as "Unhandled event loop exception" **error**. A warning, if any, makes sense.

I assume org.eclipse.ui.application.WorkbenchAdvisor.eventLoopException(Throwable) should ignore or log OperationCanceledException instances, but might be there is a better idea.

```
eclipse.buildId=4.25.0.202207280832
java.version=17.0.3
java.vendor=Eclipse Adoptium
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en_US
Command-line arguments:  -data file:/tmp/wsp_rnd_smoke/ -os linux -ws gtk -arch x86_64

org.eclipse.ui
Error
Thu Jul 28 11:41:09 CEST 2022
Unhandled event loop exception

org.eclipse.core.runtime.OperationCanceledException
	at org.eclipse.ui.texteditor.AbstractTextEditor.validateState(AbstractTextEditor.java:4906)
	at org.eclipse.ui.texteditor.AbstractTextEditor.lambda$4(AbstractTextEditor.java:4947)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:74)
	at org.eclipse.ui.texteditor.AbstractTextEditor.validateEditorInputState(AbstractTextEditor.java:4947)
	at org.eclipse.ui.texteditor.StatusTextEditor.validateEditorInputState(StatusTextEditor.java:159)
	at org.eclipse.ui.texteditor.AbstractDecoratedTextEditor.validateEditorInputState(AbstractDecoratedTextEditor.java:1055)
	at org.eclipse.ui.texteditor.AbstractTextEditor$ElementStateListener$Validator.verifyText(AbstractTextEditor.java:392)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:277)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5794)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1529)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1555)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1538)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1316)
	at org.eclipse.swt.custom.StyledText.modifyContent(StyledText.java:7607)
	at org.eclipse.swt.custom.StyledText.sendKeyEvent(StyledText.java:8539)
	at org.eclipse.swt.custom.StyledText.doContent(StyledText.java:2540)
	at org.eclipse.swt.custom.StyledText.handleKey(StyledText.java:6222)
	at org.eclipse.swt.custom.StyledText.handleKeyDown(StyledText.java:6253)
```

### PR DESCRIPTION
**Pull request by jukzi**
Avoids "Unhandled event loop exception" log entry.

### CODE REVIEW

**Comment by jukzi** [comment]
@iloveeclipse please try this. - works for me (windows).

**Comment by iloveeclipse** [comment]
Well, if that really works on Windows, we can merge that. For me on RHEL 7.9 / GTK 3.22 it fixes the "unhandled event loop error" only, but the original issue is still there. Since that restores the status quo, we can take that.

**Comment by iloveeclipse** [comment]
@jukzi : please rebase on master.

**Comment by iloveeclipse** [comment]
> please rebase on master.

No need, there is a new button.

File: org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java [hunk 1]
```diff
@@ -2795,6 +2797,7 @@ protected final void setCompatibilityMode(boolean compatible) {
 	 *
 	 * @param helpContextId the help context id
 	 */
+	@Override
```
Comment (iloveeclipse) [inline org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java:2800]:
```suggestion
```

**Review by iloveeclipse — CHANGES_REQUESTED** [review]
The extra Override is unrelated.

**Review by iloveeclipse — APPROVED** [review]
LGTM

**Comment by jukzi** [comment]
Thanks Andrey for finalizing it.

**Pull request merged**